pkgs/games/redeclipse/default.nix

```diff
@@ -0,0 +1,106 @@
+    { name = "john"; sha256 = "1jdmwkrdi5b9pivkm22rxhmkk1db9dx6l54wzcc23cvdz04ij93k"; }
+    { name = "jojo"; sha256 = "0f7kjy43fbk9kw8fip6bbw4gn3pryh0fndlahjfkaysrx98krdj3"; }
+    { name = "jwin"; sha256 = "0nc8dndnpqk2ad6316a8k6kgzsrkpwvk8s4gyh7aqfi4axfclril"; }
+    { name = "luckystrike"; sha256 = "04jiipqahphmvz5cd74dygr62dlvv6l4iglb8hzh4pp8frhls8bq"; }
+    { name = "maps"; sha256 = "0an46ipjvw4h0nxvb6qvnzp1cdkzlkiinqz4zh9lmxy1ds0gclim"; }
+    { name = "mayhem"; sha256 = "15k10imm2wr6c6fq35n4r99k7kz7n9zdnjlw6dmdq6fba67i6sbc"; }
+    { name = "mikeplus64"; sha256 = "0v4wiiivm3829j4phlavy22n4c6k6ib9ixxpdz7r6ysg5cdkaw33"; }
+    { name = "misc"; sha256 = "13rfgwrlfhflz6inbkg3fypyf8im0m49sw112b17qrw2zgp8i1rz"; }
+    { name = "nieb"; sha256 = "0z0h9jdn2gkkjil3vsvwidb1p2k09pi5n3wjxza12hhvqmcs7q8f"; }
+    { name = "nobiax"; sha256 = "08bfp4q6gbfis18bp1h4d0hqssk79jc4fhyjxnv21dbam4v4mnkn"; }
+    { name = "particles"; sha256 = "1vsx3fgg19xggxfhz3vlrh6nqhmw7kl9kmxrvb2j84blp00vd6z2"; }
+    { name = "philipk"; sha256 = "14irscg80607i5k5l2ci0n9nwibvda2f3xsykgv96d0vldrp5n5a"; }
+    { name = "projectiles"; sha256 = "09bnfsrywirwgjm6d7ff5nicx6w6b7w9568x9pb5q0ncaps3l85s"; }
+    { name = "props"; sha256 = "1dlabbprlkif8af3daf9nbgcwgxiymvj0yiphqhlri8ylfy2vpz4"; }
+    { name = "skyboxes"; sha256 = "14bi3md5y47cvb9ybipdvksz40gqsqw2r0lh3zzqb4acq367w18y"; }
+    { name = "snipergoth"; sha256 = "0m8rvvy5n8n9pm0b5cqvzsxsw51mqk8m7s1h3qc849b38isliwq2"; }
+    { name = "sounds"; sha256 = "0ivf3w5bphz5pzzx6kwcb67vbly1l19cgv3s0cyp8n87afiqj5rd"; }
+    { name = "textures"; sha256 = "0qdmgx7zbcqnb9rrga2izr93p5inirczhddfxs504rsnv0v8vyxm"; }
+    { name = "torley"; sha256 = "05ppyhghq859cbbxzj3dnl9fcx3ghy04ds1pylypwg2hsxzbjwcd"; }
+    { name = "trak"; sha256 = "0g3vq86q91a3syli38lwc8ca4ychfwsmmqf85kqzfzyd627ybclm"; }
+    { name = "ulukai"; sha256 = "0asa5fz400impklcg6dy2f7jiaqfc1sn1c36fpg8jd01gw66lw93"; }
+    { name = "unnamed"; sha256 = "0rz5683j7sfwkcycfypbv4b0ihp0qwn9rzskfsabwc1s5g324917"; }
+    { name = "vanities"; sha256 = "13f18783rc8cjf22p61zr8m5g1migzlx05fzl8xnbjdkqq4cdyix"; }
+    { name = "vegetation"; sha256 = "1y5d97nfmvax7y4fr0y5v0c8zb1ajkqwx76kjd4qc9n4spdsi5sc"; }
+    { name = "weapons"; sha256 = "103g1dhxv5ffz4ddg2xcbshbgv9606chsbas3pzk6h9ybqsyjrqh"; }
+    { name = "wicked"; sha256 = "1884rk34a2dj83gz82rc4zh3ch0dyj5221hvsr0a5h60578i7yj6"; }
+  ];
+in
+
+stdenv.mkDerivation rec {
+  pname = "redeclipse";
+  version = "1.6.0";
+
+  srcs = map (smod:
+    (fetchFromGitHub {
+      owner = "red-eclipse";
+      name = "${smod.name}";
+      repo = "${smod.name}";
+      rev = "v${version}";
+      sha256 = "${smod.sha256}";
+    })) submodules;
+
+  buildInputs = [
+    curl ed pkgconfig libX11 zlib
+    SDL2 SDL2_image SDL2_mixer
+  ];
+
+  sourceRoot = "base";
+
+  makeFlags = [ "-C src/" "prefix=$(out)" ];
+
+  patchPhase = ''
+    # -ffast-math  triggers a bug in the gamma function.
```

Comment:
What exactly do you mean?

Reply:
Enabling -ffast-math triggers a bug that traces back to `glibc-2.27-dev/include/bits/math-finite` about `__gammal_r_finite` not being defined. Example output is
```/nix/store/yjdnj9pcjgb09xif4ngmf13vb8npklys-glibc-2.27-dev/include/bits/math-finite.h:187:21: note: suggested alternative: '__gamma_r_finite'
In file included from /nix/store/6a1djj4dvsd8bdvs544krj64zlq2d1xf-gcc-7.4.0/include/c++/7.4.0/cmath:45:0,
                 from /nix/store/6a1djj4dvsd8bdvs544krj64zlq2d1xf-gcc-7.4.0/include/c++/7.4.0/math.h:36,
                 from shared/cube.h:13,
                 from engine/engine.h:5,
                 from game/game.h:4,
                 from game/ai.cpp:1:
/nix/store/yjdnj9pcjgb09xif4ngmf13vb8npklys-glibc-2.27-dev/include/bits/math-finite.h: In function '_Float128 tgammaf128(_Float128)':
/nix/store/yjdnj9pcjgb09xif4ngmf13vb8npklys-glibc-2.27-dev/include/bits/math-finite.h:187:21: error: '____gammaf128_r_finite' was not declared in this scope
   _Mdouble_ __res = __REDIRTO (gamma, _r) (__d, &__local_signgam);```

Reply:
It's fixed if you apply this as a patch: https://github.com/red-eclipse/base/commit/b16b4963c1ad81bb9ef784bc4913a4c8ab5f1bb4

Seems like they haven't made a release since. But it's fixed on their master.

```diff
diff --git c/pkgs/games/redeclipse/default.nix i/pkgs/games/redeclipse/default.nix                       
index cf4e974c04a..9587daba7b3 100644                                                                    
--- c/pkgs/games/redeclipse/default.nix                                                                  
+++ i/pkgs/games/redeclipse/default.nix                                                                  
@@ -1,4 +1,4 @@                                                                                          
-{ stdenv, fetchFromGitHub                                                                               
+{ stdenv, fetchFromGitHub, fetchurl                                                                     
 , curl, ed, pkgconfig, zlib, libX11                                                                     
 , SDL2, SDL2_image, SDL2_mixer                                                                          
 }:                                                                                                      
@@ -68,15 +68,19 @@ stdenv.mkDerivation rec {                                                            
     SDL2 SDL2_image SDL2_mixer                                                                          
   ];                                                                                                    
                                                                                                         
+  patches = [                                                                                           
+    # "remove gamma name hack" - Can't find `____gammaf128_r_finite` otherwise                          
+    # Is likely to be included in next release                                                          
+    (fetchurl {                                                                                         
+      url = "https://github.com/red-eclipse/base/commit/b16b4963c1ad81bb9ef784bc4913a4c8ab5f1bb4.diff"; 
+      sha256 = "099ci77c67pyznjqrfsr14qnfc5fg7vccl6mldxcafnwiqw0wk3m";                                  
+    })                                                                                                  
+  ];                                                                                                    
+                                                                                                        
   sourceRoot = "base";                                                                                  
                                                                                                         
   makeFlags = [ "-C src/" "prefix=$(out)" ];                                                            
                                                                                                         
-  patchPhase = ''                                                                                       
-    # -ffast-math  triggers a bug in the gamma function.                                                
-    sed -i 's/ -ffast-math//' src/Makefile                                                              
-  '';                                                                                                   
-                                                                                                        
```

Reply:
I've included the patch.